Content.Server/Electrocution/ElectrocutionSystem.cs

```diff
@@ -23,107 +23,137 @@
+
+    private const float ParalyzeTimeMultiplier = 1f;
+
+    private const float StutteringTimeMultiplier = 1.5f;
+
+    private const float JitterTimeMultiplier = 0.75f;
+    private const float JitterAmplitude = 80f;
+    private const float JitterFrequency = 8f;
+
+    public override void Initialize()
     {
-        [Dependency] private readonly EntityLookupSystem _entityLookup = default!;
-        [Dependency] private readonly IRobustRandom _random = default!;
-        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
-        [Dependency] private readonly StatusEffectsSystem _statusEffectsSystem = default!;
-        [Dependency] private readonly SharedJitteringSystem _jitteringSystem = default!;
-        [Dependency] private readonly SharedStunSystem _stunSystem = default!;
-        [Dependency] private readonly SharedStutteringSystem _stutteringSystem = default!;
-        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
-        [Dependency] private readonly DamageableSystem _damageableSystem = default!;
-        [Dependency] private readonly NodeGroupSystem _nodeGroupSystem = default!;
-        [Dependency] private readonly IAdminLogManager _adminLogger= default!;
-        [Dependency] private readonly TagSystem _tagSystem = default!;
-
-        private const string StatusEffectKey = "Electrocution";
-        private const string DamageType = "Shock";
-
-        // Yes, this is absurdly small for a reason.
-        private const float ElectrifiedDamagePerWatt = 0.0015f;
+        base.Initialize();
 
-        private const float RecursiveDamageMultiplier = 0.75f;
-        private const float RecursiveTimeMultiplier = 0.8f;
+        SubscribeLocalEvent<ElectrifiedComponent, StartCollideEvent>(OnElectrifiedStartCollide);
+        SubscribeLocalEvent<ElectrifiedComponent, AttackedEvent>(OnElectrifiedAttacked);
+        SubscribeLocalEvent<ElectrifiedComponent, InteractHandEvent>(OnElectrifiedHandInteract);
+        SubscribeLocalEvent<ElectrifiedComponent, InteractUsingEvent>(OnElectrifiedInteractUsing);
+        SubscribeLocalEvent<RandomInsulationComponent, MapInitEvent>(OnRandomInsulationMapInit);
 
-        private const float ParalyzeTimeMultiplier = 1f;
-
-        private const float StutteringTimeMultiplier = 1.5f;
+        UpdatesAfter.Add(typeof(PowerNetSystem));
+    }
 
-        private const float JitterTimeMultiplier = 0.75f;
-        private const float JitterAmplitude = 80f;
-        private const float JitterFrequency = 8f;
+    public override void Update(float frameTime)
+    {
+        UpdateElectrocutions(frameTime);
+        UpdateState(frameTime);
+    }
 
-        public override void Initialize()
+    private void UpdateElectrocutions(float frameTime)
+    {
+        var query = AllEntityQuery<ElectrocutionComponent, PowerConsumerComponent>();
+        while (query.MoveNext(out var uid, out var electrocution, out var consumer))
```

Comment:
```suggestion
        var query = EntityQueryEnumerator<ElectrocutionComponent, PowerConsumerComponent>();
        while (query.MoveNext(out var uid, out var electrocution, out var consumer))
```

This shouldn't be enumerating pause entities.